components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.h

```diff
@@ -16,6 +16,11 @@ namespace brave_wallet {
 
 class NetworkManager;
 
+struct PolkadotBlockHeader {
+  std::array<uint8_t, kPolkadotBlockHashSize> parent_hash = {};
+  uint32_t number = -1;
```

Comment:
Is `-1` being used as a sentinel for something?

Reply:
There is a lot of such tiny data structs in wallet to represent some data came from RPC. These structs are typically zero initalized and don't introduce sentinel values. Having dedicated sentinel value means we always have to add boilerplate checks for field if it matches that sentinel. 
Instead we just assume that if isntance of this data stuct appears somewhere in the code then it was initialized with values from RPC, i.e. having instance of such struct itself means that "sentinel" check was passed.

So I think we should have this struct zero initialized.